Core/Core/Submissions/Submission.swift

```diff
@@ -333,21 +333,81 @@ extension Submission {
         return excused == true || (score != nil && workflowState == .graded)
     }
 
-    /// Returns the appropriate status text.
+    public var status: SubmissionStatus {
+        if late { return .late }
+        if missing { return .missing }
+        if submittedAt != nil { return .submitted }
+        return .notSubmitted
+    }
+
+    /// Returns the appropriate status description considering onPaper & noSubmission scenarios.
     /// If the submission has been submitted and it's graded already, then it returns "Graded".
     /// Otherwise it returns the submissions's status.
     /// Graded submissions that have been resubmitted will return "Submitted".
-    public var statusText: String {
-        status == .submitted && needsGrading == false
-            ? String(localized: "Graded", bundle: .core)
-            : status.text
+    public var statusDescription: SubmissionStatusDescription {
+        let desc: SubmissionStatusDescription = {
+            if case .notSubmitted = status {
+                if let submissionTypes = assignment?.submissionTypes {
+                    if submissionTypes.contains(.on_paper) { return .onPaper }
+                    if submissionTypes.contains(.none) { return .noSubmission }
+                }
+                return .byStatus(.notSubmitted)
+            } else {
+                return .byStatus(status)
+            }
+        }()
+
+        // Graded check
+        switch desc {
+        case .byStatus(.submitted):
+            return needsGrading == false ? .graded : desc // Maintaining the old logic
+        case .onPaper, .noSubmission:
+            return isGraded ? .graded : desc
+        default:
+            return desc
+        }
     }
+}
 
-    public var status: SubmissionStatus {
-        if late { return .late }
-        if missing { return .missing }
-        if submittedAt != nil { return .submitted}
-        return .notSubmitted
+public enum SubmissionStatusDescription: Equatable {
+    case byStatus(SubmissionStatus)
+    case onPaper
+    case noSubmission
+    case graded
+
+    public var text: String {
+        switch self {
+        case .byStatus(let status):
+            return status.text
+        case .onPaper:
+            return String(localized: "On Paper", bundle: .core)
+        case .graded:
+            return String(localized: "Graded", bundle: .core)
+        case .noSubmission:
+            return String(localized: "No Submission", bundle: .core)
+        }
+    }
+
+    public var color: UIColor {
+        switch self {
+        case .byStatus(let status):
+            return status.color
+        case .graded:
+            return .textSuccess
+        default:
```

Comment:
Please use the cases explicitly instead of `default` (and mind the ordering).
Same for `icon` below